roles/test_operator/tasks/main.yml

```diff
@@ -0,0 +1,210 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure test_operator folder exists
+  ansible.builtin.file:
+    path: "{{ cifmw_test_operator_artifacts_basedir }}"
+    state: directory
+    recurse: true
+
+- name: Setup tempest tests
+  ansible.builtin.include_tasks: tempest-tests.yml
+
+- name: Ensure OperatorGroup for the test-operator is present
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: operators.coreos.com/v1
+      kind: OperatorGroup
+      metadata:
+        name: test-operator-operatorgroup
+        namespace: "{{ cifmw_test_operator_namespace }}"
+      spec:
+        targetNamespaces:
+          - "{{ cifmw_test_operator_namespace }}"
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Ensure CatalogSource for the test-operator is present
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: operators.coreos.com/v1alpha1
+      kind: CatalogSource
+      metadata:
+        name: test-operator-catalog
+        namespace: "{{ cifmw_test_operator_namespace }}"
+      spec:
+        sourceType: grpc
+        image: "{{ cifmw_test_operator_index }}"
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Ensure Subscription for the test-operator is present
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: operators.coreos.com/v1alpha1
+      kind: Subscription
+      metadata:
+        name: test-operator
+        namespace: "{{ cifmw_test_operator_namespace }}"
+      spec:
+        name: test-operator
+        source: test-operator-catalog
+        sourceNamespace: "{{ cifmw_test_operator_namespace }}"
+  when: not cifmw_test_operator_dry_run | bool
+
+- name: Check that test-operator is running # noqa risky-shell-pipe
+  register: test_operator_status
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  until: test_operator_status.stdout is search("Running")
+  delay: 10
+  retries: 20
+  ansible.builtin.shell: |
+    oc get pods | grep {{ cifmw_test_operator_controller_name }}
```

Comment:
+1. We have a tone of examples for this. 
I'd suggest to pass the target namespace explicitly to avoid relying on the one set in kubeconfig by who knows which command (probably, the last `oc new-project`.

Reply:
> Isn' t possible to do this with kubernetes.core.k8s_info instead?

Yes, it is possible. I just couldn't figure out how to do a partial match via the k8s_info at first. 

> I'd suggest to pass the target namespace explicitly to avoid relying on the one set in kubeconfig by who knows which command (probably, the last oc new-project.

Done. Thank you!